src/main/java/io/github/eatmyvenom/litematicin/utils/Printer.java

```diff
@@ -495,10 +505,40 @@ public static ActionResult doPrinterAction(MinecraftClient mc) {
-
+                            Block sBlock = stateSchematic.getBlock();
                         if (stateSchematic == stateClient) {
                             continue;
-                        }
+                        } else if (sBlock instanceof SandBlock) {
+			BlockPos Offsetpos = new BlockPos(x, y-1, z);
+                    		BlockState OffsetstateSchematic = world.getBlockState(Offsetpos);
+                    		BlockState OffsetstateClient = mc.world.getBlockState(Offsetpos);
+                                        if (OffsetstateClient.isAir() || (breakBlocks && !OffsetstateClient.getBlock().getName().equals(OffsetstateSchematic.getBlock().getName())) )
+                                            continue;
+                                    } else if (sBlock instanceof ConcretePowderBlock) {
+			BlockPos Offsetpos = new BlockPos(x, y-1, z);
+                    		BlockState OffsetstateSchematic = world.getBlockState(Offsetpos);
+                    		BlockState OffsetstateClient = mc.world.getBlockState(Offsetpos);
+                                        if (OffsetstateClient.isAir() || (breakBlocks && !OffsetstateClient.getBlock().getName().equals(OffsetstateSchematic.getBlock().getName())) )
+                                            continue;
+                                    } else if (sBlock instanceof GravelBlock) {
+			BlockPos Offsetpos = new BlockPos(x, y-1, z);
+                    		BlockState OffsetstateSchematic = world.getBlockState(Offsetpos);
+                    		BlockState OffsetstateClient = mc.world.getBlockState(Offsetpos);
+                                        if (OffsetstateClient.isAir() || (breakBlocks && !OffsetstateClient.getBlock().getName().equals(OffsetstateSchematic.getBlock().getName())) )
+                                            continue;
+                                    } else if (sBlock instanceof AnvilBlock) {
+			BlockPos Offsetpos = new BlockPos(x, y-1, z);
+                    		BlockState OffsetstateSchematic = world.getBlockState(Offsetpos);
+                    		BlockState OffsetstateClient = mc.world.getBlockState(Offsetpos);
+                                        if (OffsetstateClient.isAir() || (breakBlocks && !OffsetstateClient.getBlock().getName().equals(OffsetstateSchematic.getBlock().getName())) )
+                                            continue;
+                                    } else if (sBlock instanceof DragonEggBlock) {
+			BlockPos Offsetpos = new BlockPos(x, y-1, z);
+                    		BlockState OffsetstateSchematic = world.getBlockState(Offsetpos);
+                    		BlockState OffsetstateClient = mc.world.getBlockState(Offsetpos);
+                                        if (OffsetstateClient.isAir() || (breakBlocks && !OffsetstateClient.getBlock().getName().equals(OffsetstateSchematic.getBlock().getName())) )
+                                            continue;
+                                    }
```

Comment:
Here you write the same thing multiple times, this fits in one if statement.
```suggestion
                        } else if (sBlock instanceof SandBlock || sBlock instanceof ConcretePowderBlock 
                        || sBlock instanceof GravelBlock || sBlock instanceof AnvilBlock || sBlock instanceof DragonEggBlock) {
			BlockPos Offsetpos = new BlockPos(x, y-1, z);
                    		BlockState OffsetstateSchematic = world.getBlockState(Offsetpos);
                    		BlockState OffsetstateClient = mc.world.getBlockState(Offsetpos);
                                        if (OffsetstateClient.isAir() || (breakBlocks && !OffsetstateClient.getBlock().getName().equals(OffsetstateSchematic.getBlock().getName())) )
                                            continue;
                                    } 
```